visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts

```diff
@@ -46,7 +45,6 @@ export class CodeMapMouseEventService
 
 	private mouse: Coordinates = { x: 0, y: 0 }
 	private oldMouse: Coordinates = { x: 0, y: 0 }
-	private intersectionResult: IntersectionResult
```

Comment:
We need to track the intersectionResult for this line below `if (highlightedBuilding && this.intersectionResult.intersectionFound)`.  This change was required, because we can now have highlighted buildings without an intersection by hovering over the distribution-bar.

Reply:
That still works as in fact we use the same resource `highlighted` (buildings) inside the threeSceneService. When hovering the fileExtenseionBar it adds the buildings into that list and this list is used from the mouseEventService. So we don't have to duplicate and store the highlightedBuildings inside the mousEventService.

Reply:
My bad. This was related to another issue that reoccurs now: When you select a building all buildings that are connected to it get highlighted. When you click on an empty space to deselect, there are still buildings hovered. That's why we have to check the intersectionResult later on again.

Reproduce:
- Load a map with edges
- Select an edge metric
- Select a building
- Try deselecting it by clicking into the void

Reply:
uhh that is weird. I'll have a look

Reply:
I was now adapting the functionality as before. I think this should be good enough for now until the hovered/selected buildings are somehow stored in our redux state.